### PR TITLE
WIP github-ci: add dpdk codecov build

### PR DESCRIPTION
**Pull request by victorjulien**
Experiment to see if I can get codecov to cover the dpdk "live" logic.

### CODE REVIEW

File: .github/workflows/builds.yml [hunk 1]
```diff
@@ -1472,6 +1472,167 @@ jobs:
...
+                zlib1g-dev \
+                exuberant-ctags \
+                unzip \
+                curl \
+                time \
+                wget \
+                dpdk-dev
+      # packaged Rust version is too old for coverage, so get from rustup
+      - name: Install Rust
+        run: curl https://sh.rustup.rs -sSf | sh -s -- --default-toolchain 1.63.0 -y
```
Comment (jasonish) [inline .github/workflows/builds.yml:1540]: Packaged version is 1.75.0 now.

File: .github/workflows/builds.yml [hunk 2]
```diff
@@ -1472,6 +1472,167 @@ jobs:
...
+      - uses: actions/download-artifact@c850b930e6ba138125429b7e5c93fc707a7f8427
+        with:
+          name: cbindgen
+          path: prep
+      - name: Setup cbindgen
+        run: |
+          mkdir -p $HOME/.cargo/bin
+          cp prep/cbindgen $HOME/.cargo/bin
+          chmod 755 $HOME/.cargo/bin/cbindgen
+          echo "$HOME/.cargo/bin" >> $GITHUB_PATH
```
Comment (jasonish) [inline .github/workflows/builds.yml:1557]: Use package as well.

File: .github/workflows/builds.yml (hunk 1 above)
Comment (victorjulien) [reply .github/workflows/builds.yml:1540]: IIRC with coverage it's important to make sure the llvm tool chain and rust match up, so will check that later

**Comment by victorjulien** [comment]
replaced by #10949

**Pull request closed without merging**